controls/problems.js

```diff
@@ -20,7 +60,49 @@ helper.submitSolution = async (req, res, next) => {
         }
         var results = await checkAnswer(data);
 
-        //add code to attach this submissions data to user's account
+        //code to attach this submissions data to user's account
+        var tcs = [], verdict = 'Accepted', time=0, mem=0, flag = false;
+        for (i=0; i<results.length; i++){
+            time = Math.max(time, results[i].time);
+            mem = Math.max(mem, results[i].memory);
+            if (flag === false && results[i].status.description !== 'Accepted'){
+                verdict = results[i].status.description;
+                flag = true;
+            }
+            tcs.push({
+                status: results[i].status.description,
+                time: results[i].time,
+                memory: results[i].memory
+            });
+        }
+        var langName;
+        const subCount = await submission.countDocuments({});
+        for (var i=0; i<lang.length; i++){
+            if (lang[i].id === req.body.language){
+                langName = lang[i].name;
+                break; 
+            }
+        }
+        var newSubmission = new submission({
+            username: req.user ? req.user.username : 'Guest',
+            qID: req.body.qID,
+            subID: 1 + subCount,
+            code: req.body.code,
+            language: langName,
```

Comment:
`langName` is not being inserted into the database. Also, there are fewer compilers available now.
One easy thing you could do was either make `lang.json` with all the compilers or let all the compilers be in .ejs itself, but add `data.langCompiler = $("#language option:selected").text();` to `./public/js/solution_submit.js` and directly use `req.body.langCompiler` in `./controls/problems.js`

Reply:
please review again @Vishal1541